### PR TITLE
kinetis: adaption for common makefiles

### PR DESCRIPTION
**Pull request by jfischer-phytec-iot**
like "refactor board makefiles #2852"

@gebart Do you agree with 7704626 and you can test it please? I did not dare to change the Mulle's Makefile :-).

~~3150af0 is also necessary because not every Cortex-M4 has a FPU.~~

### CODE REVIEW

**Comment by haukepetersen** [comment]
nice. The FPU issue is fixed in #3023. So I would suggest to merge #3023 first and adapt this...

**Comment by jfischer-phytec-iot** [comment]
@haukepetersen ok, will rebase/refactor it after #3023 is merged.

**Comment by gebart** [comment]
@jfischer-phytec-iot I will test this asap. You may delete all of the devio stuff as well I guess.

**Comment by jfischer-phytec-iot** [comment]
rebased on #3023

**Comment by jfischer-phytec-iot** [comment]
rebased on master

**Comment by gebart** [comment]
#3100 is merged now. Please rebase.

**Comment by jfischer-phytec-iot** [comment]
@gebart rebased, please review and test it. I had a strange behavior of I2C if heap is placed at the end of ram. I initially left it as it was.

**Comment by gebart** [comment]
I have some fixes for mulle, I have opened a PR in your repo at https://github.com/jfischer-phytec-iot/RIOT/pull/15

**Comment by gebart** [comment]
Did you fix the i2c issue?
I don't have any i2c devices with me right now to test with, but I can probably do it on Monday.

File: cpu/kinetis_common/ldscripts/kinetis-base.ld [hunk 1]
```diff
@@ -227,6 +227,15 @@ SECTIONS
         _estack = .;
     } > sram_u
 
+    /* heap section */
```
Comment (gebart) [inline cpu/kinetis_common/ldscripts/kinetis-base.ld]: This seem like a left over after your latest rebase.

**Comment by jfischer-phytec-iot** [comment]
@gebart I do not have found a issue. If the heap shifted backwards, then hdc1000 sernsor no longer works (also on the FRDM k64f-board), everything else works. I have no way to measure on i2c until Monday :-)

File: cpu/kinetis_common/ldscripts/kinetis-base.ld (hunk 1 above)
Comment (jfischer-phytec-iot) [inline cpu/kinetis_common/ldscripts/kinetis-base.ld]: I want to keep it for the time being.

**Comment by jfischer-phytec-iot** [comment]
I mark it WIP (It only applies to c20ef10) until I checked i2c on Monday.

**Comment by gebart** [comment]
A hint: examine the .map file and see if any .data or .bss symbols overlap
with the sheap - eheap area.

**Comment by jfischer-phytec-iot** [comment]
@gebart Mulle build was broken, look at afe1788, it was necessary to move linkerscripts because they are hardcoded to $(RIOTCPU)/$(CPU)/$(CPU_MODEL)_linkersscript.ld in boards/Makefile.include.cortexm_common.

**Comment by gebart** [comment]
@jfischer-phytec-iot rebase on latest master since #3120, like I wrote in the original PR in your repo, "Depends on RIOT-OS#3120"

**Comment by jfischer-phytec-iot** [comment]
@gebart Sorry, got it too late, will revert and rebase

**Comment by jfischer-phytec-iot** [comment]
rebased

**Comment by jfischer-phytec-iot** [comment]
@gebart I have found the reason for the anomaly, #3124. Ready for review.

**Comment by gebart** [comment]
Nice, I will review and test this on mulle tomorrow.

**Comment by gebart** [comment]
Works fine except for make term, I forgot to add the proper include. I opened https://github.com/jfischer-phytec-iot/RIOT/pull/16

**Comment by gebart** [comment]
ACK, please squash all commits marked "squash"

**Comment by jfischer-phytec-iot** [comment]
@jremmert-phytec-iot  We had that before, what was it ? -> failed: driver_kw2xrf, driver_xbee

**Comment by jfischer-phytec-iot** [comment]
ups

**Comment by gebart** [comment]
the error message is:
```
riot/sys/auto_init/netif/auto_init_ng_at86rf2xx.c:27:33: fatal error: ng_at86rf2xx_params.h: No such file or directory
 #include "ng_at86rf2xx_params.h"
```
so it seems like a problem with the ng_at86rf2xx module

**Comment by gebart** [comment]
I think we can just blacklist the xbee and kw2xrf tests for mulle for the time being until I have had time to properly try out the ng_ stack. The kw2xrf driver (not sure about xbee) is not applicable to the mulle platform anyway.

**Comment by jfischer-phytec-iot** [comment]
>The kw2xrf driver (not sure about xbee) is not applicable to the mulle platform anyway.

@gebart maybe yes : http://www.freescale.com/webapp/sps/site/prod_summary.jsp?code=MCR20A
This is the same modem.

**Comment by jremmert-phytec-iot** [comment]
> @jremmert-phytec-iot We had that before, what was it ? -> failed: driver_kw2xrf, driver_xbee

Without having a deeper look into it, the issues we had before were:
 1. The Makefile of the test application for the radio-drivers defines default GPIO-pins if the Board did not define pins for that radio module (I think the pins are counted for GPIO0 to 4 or so).

 2. And the second problem. The driver module was missing in some Makefile (I guess it was Makefile.include) that the driver was not .

**Comment by jremmert-phytec-iot** [comment]
@gebart you need to define the pins (SPI, IRQ-pin...) for the radio-driver you want to use. E.g. for the samr21-xpro board these pin assignments are defined in ```./boards/samr21-xpro/include/ng_at86rf2xx_params.h```

The makefile in ```./boards/mulle/Makfile.dep``` tells to use the ng_at... driver. Without that module definition it should build either.

**Comment by gebart** [comment]
@jremmert-phytec-iot Thank you for the explanation. I think I will make a separate PR with the ng_netif change..

@jfischer-phytec-iot could you remove https://github.com/jfischer-phytec-iot/RIOT/commit/1d06ddebaebbe2f53c9db7b2aedb3ba534659408 (mulle: Add dependency check for ng_netif) from this PR?

I'll open a specific PR for the transceiver settings for mulle.

**Comment by jfischer-phytec-iot** [comment]
@gebart removed 1d06dde, squashed and rebased on master

**Comment by gebart** [comment]
ACK
@haukepetersen do you want to say anything or can we merge this so that all Cortex M platforms are finally aligned with regards to Makefiles and linker?

**Comment by jfischer-phytec-iot** [comment]
broadcast: Can we merge it?

**Comment by kaspar030** [comment]
Yeah, merge!

**Pull request merged**